src/lib/cli_lib/commands.ml

```diff
@@ -140,3 +140,199 @@ let validate_transaction =
+                      let open Consensus_vrf.Layout in
+                      let message =
+                        Result.ok_or_failwith (Message.of_yojson message_json)
+                      in
+                      let evaluation =
+                        Evaluation.of_message_and_sk ~constraint_constants
+                          message keypair.private_key
+                      in
+                      Format.printf "%a@."
+                        (Yojson.Safe.pretty_print ?std:None)
+                        (Evaluation.to_yojson evaluation) ;
+                      Deferred.return (`Repeat ())
+                    with Yojson.End_of_input -> return (`Finished ()) )
+                >>| function
+                | Ok x ->
+                    x
+                | Error err ->
+                    Format.eprintf "Error:@.%s@.@."
+                      (Yojson.Safe.pretty_to_string
+                         (Error_json.error_to_yojson err)) ;
+                    `Repeat () )
+        | Error err ->
+            eprintf "Could not read the specified keypair: %s\n"
+              (Secrets.Privkey_error.to_string err) ;
+            exit 1
+      in
+      exit 0)
+
+  let batch_check_witness =
+    Command.async
+      ~summary:"Check a batch of vrf evaluation witnesses read on stdin"
+      ( Command.Param.return @@ Exceptions.handle_nicely
+      @@ fun () ->
+      let open Deferred.Let_syntax in
+      (* TODO-someday: constraint constants from config file. *)
+      let constraint_constants =
+        Genesis_constants.Constraint_constants.compiled
+      in
+      let lexbuf = Lexing.from_channel In_channel.stdin in
+      let lexer = Yojson.init_lexer () in
+      let%bind () =
+        Deferred.repeat_until_finished () (fun () ->
+            Deferred.Or_error.try_with (fun () ->
+                try
+                  let evaluation_json =
+                    Yojson.Safe.from_lexbuf ~stream:true lexer lexbuf
+                  in
+                  let open Consensus_vrf.Layout in
+                  let evaluation =
+                    Result.ok_or_failwith
+                      (Evaluation.of_yojson evaluation_json)
+                  in
+                  let evaluation =
+                    Evaluation.compute_vrf ~constraint_constants evaluation
+                  in
+                  Format.printf "%a@."
+                    (Yojson.Safe.pretty_print ?std:None)
+                    (Evaluation.to_yojson evaluation) ;
+                  Deferred.return (`Repeat ())
+                with Yojson.End_of_input -> return (`Finished ()) )
+            >>| function
+            | Ok x ->
+                x
+            | Error err ->
+                Format.eprintf "Error:@.%s@.@."
+                  (Yojson.Safe.pretty_to_string
+                     (Error_json.error_to_yojson err)) ;
+                `Repeat () )
+      in
+      exit 0 )
+
+  let command_group =
+    Command.group ~summary:"Commands for vrf evaluations"
```

Comment:
In the summary here (and on a relevant section of the GraphQL doc too) can you include more information on exactly how one could use these commands in practice -- I think these two scenarios cover all the bases (but up to you, if you want to choose different examples here) (1) figure out all of the slots that I will win for this current epoch or (2) for some epoch, ingest a 3rd party's batch witness and report which slots they have won.

We can then digest these docs into a page on the website too, but I think it would be really great if this lived in the help message and the GraphQL schema docs too